pkg/controller/che/che_controller.go

```diff
@@ -470,20 +477,19 @@ func (r *ReconcileChe) Reconcile(request reconcile.Request) (reconcile.Result, e
 		return nil, nil
 	}
 
+	pluginRegistryURL := instance.Spec.Server.PluginRegistryUrl
 	// Create Plugin registry resources unless an external registry is used
 	externalPluginRegistry := instance.Spec.Server.ExternalPluginRegistry
 	if !externalPluginRegistry {
-		pluginRegistryURL, err := addRegistryRoute("plugin")
+		guessedPluginRegistryURL, err := addRegistryRoute("plugin")
 		if err != nil {
 			return reconcile.Result{}, err
 		}
 		if cheFlavor != "codeready" {
-			pluginRegistryURL += "/v3"
+			guessedPluginRegistryURL += "/v3"
 		}
-		instance.Status.PluginRegistryURL = pluginRegistryURL
-		if err := r.UpdateCheCRStatus(instance, "status: Plugin Registry URL", pluginRegistryURL); err != nil {
-			instance, _ = r.GetCR(request)
-			return reconcile.Result{Requeue: true, RequeueAfter: time.Second * 1}, err
+		if pluginRegistryURL == "" {
+			pluginRegistryURL = guessedPluginRegistryURL
```

Comment:
Does it matter that you're no longer throwing a status report here?

Reply:
The status report is thrown in the same way, but just some lines later, where this block has indeed been moved. because we have to update the registry URL on the status in both cases (when using embedded registry or when an external registry URL is specified)